src/Bicep.Core/Syntax/TestDeclarationSyntax.cs

```diff
@@ -53,20 +56,151 @@ this.Value switch
                 // blocked by assert in the constructor
                 _ => throw new NotImplementedException($"Unexpected type of test value '{this.Value.GetType().Name}'.")
             };
-            
+
         public ObjectSyntax GetBody() =>
             this.TryGetBody() ?? throw new InvalidOperationException($"A valid test body is not available on this test due to errors. Use {nameof(TryGetBody)}() instead.");
 
-        public ObjectSyntax? TryGetParameters(){
+        public JToken? TryGetParameters(){
             var body = this.GetBody();
             foreach (var property in body.Properties) {
                 if (property.TryGetKeyText() == "params" && property.Value is ObjectSyntax paramsObject) 
                 {
-                    return paramsObject;
+                    var parameters = new JObject();
+                    foreach (var parameter in paramsObject.Properties){
+                        var propertyName = parameter.TryGetKeyText();
+                        if (propertyName is null)
+                        {
+                            throw new NotImplementedException($"Error reading the parameter key");
+                        }
+                        else
+                        {
+                            switch (parameter.Value)
+                            {
+                                case StringSyntax stringSyntax:
+                                    parameters.Add(new JProperty(propertyName, stringSyntax.TryGetLiteralValue()));
+                                    break;
+
+                                case IntegerLiteralSyntax numericSyntax:
+                                    parameters[propertyName] = numericSyntax.Value;
+                                    break;
+
+                                case BooleanLiteralSyntax booleanSyntax:
+                                    parameters[propertyName] = booleanSyntax.Value;
+                                    break;
+                                
+                                case null:
+                                    parameters[propertyName] = JValue.CreateNull();
+                                    break;
+
+                                case ObjectSyntax objectSyntax:
+                                    parameters[propertyName] = ParseObjectSyntax(objectSyntax);
+                                    break;
+
+                                case ArraySyntax arraySyntax:
+                                    parameters[propertyName] = ParseArraySyntax(arraySyntax);
+                                    break;
+                                default:
+                                    throw new NotImplementedException($"Unexpected type of parameter value '{parameter.Value.GetType().Name}'.");
```

Comment:
We will need to generate and emit diagnostics for the error cases in the class, but that can be done in a seperate PR.